fe/fe-core/src/main/cup/sql_parser.cup

```diff
@@ -4847,8 +4847,6 @@ type ::=
   {: RESULT = ScalarType.createVarcharType(-1); :}
   | KW_ARRAY LESSTHAN type:value_type GREATERTHAN
   {: RESULT = new ArrayType(value_type); :}
-  | KW_ARRAY LESSTHAN KW_NOT_NULL LPAREN type:value_type RPAREN GREATERTHAN
-  {: RESULT = new ArrayType(value_type, false); :}
```

Comment:
I think we can keep this syntax but just throw an exception due to we don't support it temporarily.

Reply:
Lots of codes who called `new ArrayType` will also need to catch the exception,  if we throw an exception while new ArrayType. I think just remove these too lines is better.